pyqg/qg_model.py

```diff
@@ -86,20 +86,20 @@ def __init__(
         U2 : number
             Lower layer flow. Units: meters seconds :sup:`-1`
         """
+        if 'nz' in kwargs:
+            del kwargs['nz']
```

Comment:
Why? Can we avoid this? Can we just remove the `**kwargs` argument completely?

Reply:
This got added in an earlier PR to facilitate reinitializing a new model with the same configuration parameters, but we can avoid it in a much better way. I'll make sure it's removed. I don't think removing `**kwargs` makes as much sense though, since we have them in every other model and it's nice to avoid needing to respecify all the parameters of the parent class.

Reply:
Alright, so it looks like the `delegate` changes play poorly here with the `_config` method (which is used in that re-initialization procedure I just alluded to), but also that method just needs a lot more testing. I might add some additional tests to this pull request because the existing ones did not pick up the difference.

Reply:
Ok, so we can resolve that issue by removing the spurious `__init__` redefinition from delegate, but this is turning into a bit of a can of worms. I think we should actually start off by removing the existing `subgrid_forcing` method and all associated methods (e.g. `_config`), maybe treating it as a patch-level update because it's basically a bug fix. Then we rebase this on top of that change and everything is much simpler. We can revisit subgrid forcing / everything else in a subsequent PR.

Reply:
Here's a PR to do that: https://github.com/pyqg/pyqg/pull/308

Reply:
Alright, this is now done.